src/core/function_devportal_service_data.tf

```
@@ -0,0 +1,274 @@
+  function_app_name   = module.function_devportalservicedata.name
+  function_app_id     = module.function_devportalservicedata.id
+  app_service_plan_id = local.function_devportalservicedata.app_context.app_service_plan.id
+
+  application_insights_instrumentation_key = data.azurerm_application_insights.application_insights.instrumentation_key
+
+  storage_account_name               = module.function_devportalservicedata.storage_account_name
+  storage_account_access_key         = module.function_devportalservicedata.storage_account.primary_access_key
+  internal_storage_connection_string = module.function_devportalservicedata.storage_account_internal_function.primary_connection_string
+
+  runtime_version   = "~3"
+  os_type           = "linux"
+  health_check_path = "api/v1/info"
+
+  subnet_id = local.function_devportalservicedata.app_context.snet.id
+  allowed_ips = concat(
+    [],
+  )
+  allowed_subnets = [
+    data.azurerm_subnet.azdoa_snet[0].id,
+  ]
+
+  app_settings = merge(local.function_devportalservicedata.app_settings_commons, {
+    // disable listeners on staging slot
+    "AzureWebJobs.OnServiceChange.Disabled"                 = "1"
+    "AzureWebJobs.UpsertSubscriptionToMigrate.Disabled"     = "1"
+    "AzureWebJobs.ChangeOneSubscriptionOwnership.Disabled"  = "1"
+    "AzureWebJobs.ChangeAllSubscriptionsOwnership.Disabled" = "1"
+  })
+
+  tags = var.tags
+}
+
+#
+# DB
+#
+
+// db admin user credentials
+data "azurerm_key_vault_secret" "devportalservicedata_db_server_adm_username" {
+  name         = "devportal-servicedata-DB-ADM-USERNAME"
+  key_vault_id = data.azurerm_key_vault.common.id
+}
+data "azurerm_key_vault_secret" "devportalservicedata_db_server_adm_password" {
+  name         = "devportal-servicedata-DB-ADM-PASSWORD"
+  key_vault_id = data.azurerm_key_vault.common.id
+}
+// db applicative user credentials
+data "azurerm_key_vault_secret" "devportalservicedata_db_server_fndevportalservicedata_password" {
+  name         = "devportal-servicedata-FNDEVPORTALSERVICEDATA-PASSWORD"
+  key_vault_id = data.azurerm_key_vault.common.id
+}
+
+
+module "devportalservicedata_db_server" {
+  source = "git::https://github.com/pagopa/azurerm.git//postgres_flexible_server?ref=v2.18.10"
+
+  name                = local.function_devportalservicedata.db.name
+  location            = var.location
+  resource_group_name = local.function_devportalservicedata.app_context.resource_group.name
+
+  administrator_login    = data.azurerm_key_vault_secret.devportalservicedata_db_server_adm_username.value
+  administrator_password = data.azurerm_key_vault_secret.devportalservicedata_db_server_adm_password.value
+
+  sku_name                     = "GP_Standard_D2ds_v4"
+  db_version                   = 13
+  geo_redundant_backup_enabled = false
+
+  private_endpoint_enabled = true
+  private_dns_zone_id      = azurerm_private_dns_zone.privatelink_postgres_database_azure_com.id
+
+  high_availability_enabled = false
```

```suggestion
  high_availability_enabled = true
```
if this database is mission critical is recommended use  high availability feature

---

IMHO is not mission critical as it only provide analytics data. Anyway, it's the price difference big?

---

double

---

So I'd rather not enable high availability, as we may tolerate downtimes on this API. In case, could we enable it in future?

---

yes we can change it in the future, downtime is expected during the upgrade